sycl/include/sycl/usm.hpp

```diff
@@ -19,161 +19,316 @@ namespace sycl {
 ///
 // Explicit USM
 ///
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const device &dev, const context &ctxt,
-    const detail::code_location CodeLoc = detail::code_location::current());
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const device &dev, const context &ctxt,
-    const property_list &propList,
-    const detail::code_location CodeLoc = detail::code_location::current());
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const queue &q,
-    const detail::code_location CodeLoc = detail::code_location::current());
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const queue &q, const property_list &propList,
-    const detail::code_location CodeLoc = detail::code_location::current());
+__SYCL_EXPORT void *
+malloc_device(size_t size, const device &dev, const context &ctxt
+#ifndef DISABLE_SYCL_INSTRUMENTATION_METADATA
+              ,
+              const detail::code_location &CL = detail::code_location::current()
+#endif
```

Comment:
Can we abstract the whole thing into a single macro? Something like https://godbolt.org/z/ojG9zc81z